backend/cmd/cluster.go

```diff
@@ -13,24 +17,63 @@ import (
 const OwnClusterName = "main"
 
 type Cluster struct {
-	Name   string `json:"name"`
-	Server string `json:"server,omitempty"`
-	config *clientcmdapi.Cluster
+	Name     string `json:"name"`
+	Server   string `json:"server,omitempty"`
+	config   *clientcmdapi.Cluster
+	AuthType string `json:"auth_type"`
 }
 
-func GetOwnCluster() (*Cluster, error) {
+func GetClusterOauth2Config(clusterName string) (*OauthConfig, error) {
+	if oauth2Config, ok := oauth2ConfigsCache[clusterName]; ok {
+		return oauth2Config, nil
+	}
+
+	return nil, fmt.Errorf("couln't find auth config for cluster %s", clusterName)
+}
+
+func GetOwnCluster(headlampConfig *HeadlampConfig) (*Cluster, error) {
 	config, err := rest.InClusterConfig()
 	if err != nil {
 		return nil, err
 	}
 
+	var oauthConfig *oauth2.Config
+
+	var verifier *oidc.IDTokenVerifier
+
+	authProvider := ""
+
+	if headlampConfig.oidcIdpIssuerURL != "" {
+		authProvider = "oidc"
+		ctx := context.Background()
+
+		provider, err := oidc.NewProvider(ctx, headlampConfig.oidcIdpIssuerURL)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		oidcConfig := &oidc.Config{
+			ClientID: headlampConfig.oidcClientID,
+		}
+		verifier = provider.Verifier(oidcConfig)
+		oauthConfig = &oauth2.Config{
+			ClientID:     headlampConfig.oidcClientID,
+			ClientSecret: headlampConfig.oidcClientSecret,
+			Endpoint:     provider.Endpoint(),
+			RedirectURL:  "/oidc-callback",
+			Scopes:       []string{oidc.ScopeOpenID, "profile", "email", "groups"},
+		}
+	}
+
+	oauth2ConfigsCache[OwnClusterName] = &OauthConfig{oauthConfig, verifier, OwnClusterName}
```

Comment:
What happens when the `oauthConfig` is not correctly initialized? (i.e. if we don't have OIDC enabled)
I think we should only add the cache if we have a valid OauthConfig for it.